### ISSUE TITLE
Repeating events not displayed on first day of monthly calendar

### ISSUE THREAD

**Issue by michaelray** [nextcloud/calendar#1913]
There appears to be a regression of a display bug related to displaying repeating events on the edge of a monthly display boundary (the first date displayed on a monthly calendar view).  It only seems to be affecting repeating events, as I've got a one-time all day event on the same date and it displays, as expected.

This also seems to be a time zone-related issue when using time zones with negative UTC offsets.  My time zone is America/Chicago (UTC-6), but if I change the time zone in the calendar app settings to UTC (or other positive time zones, like Berlin (UTC+1) or Tokyo (UTC+9), the repeating events display, as expected.

In my case, I've found this occurring on Sunday, March 1 2020 for repeating events.  I've noticed that the repeating events that don't display for March 1 when looking at the month of March DO seem to display when looking at March 1 on the February view.  This, in combination with the time zone aspect mentioned above, leads me to believe that this issue is related to the web view fetching/filtering calendar events to be displayed when on the edge of a monthly display boundary.

I say that this is a regression, as this issue didn't occur with NextCloud Calendar 1.7.2, but after upgrading from that version to 2.0.1, it is occurring.

Also, I had previously reported this same exact issue quite a long time ago (Nov. 24, 2015) on the OwnCloud Calendar in https://github.com/owncloudarchive/calendar/issues/990 (this URL doesn't appear to load anymore, but it can be seen to exist on my GitHub user page contribution activity).

### Steps to reproduce
1.  Create a repeating event on a monthly display boundary (the first date displayed on a monthly calendar view).  For example, Sunday, March 1 2020 is the first date displayed on the monthly calendar for March when the calendar is showing displaying Sunday as the first day of each week (Sun-Sat).
2.  In the NextCloud calendar app settings, change the time zone to a negative offset timezone, like America/Chicago (UTC-6).
3.  View the calendar in the month view and change to show the month that shows the repeating event created on the monthly display boundary mentioned in step 1.

### Expected behaviour
The repeating events that occur on that monthly display boundary should appear.

### Actual behaviour
The repeating events that occur on that monthly display boundary do not appear.

### Calendar app
**Calendar app version:** (see apps admin page, e.g. 2.0.1)
2.0.1

**CalDAV-clients used:** (Thunderbird Lightning, DAVx5, Evolution, macOS Calendar, etc)
DAVx5

### Client configuration
**Browser:** (e.g. Firefox 48)
Firefox 72.0.2 (64-bit)

**Operating system:** (e.g. Arch Linux)
Windows 10

### Server configuration

**Operating system**: (e.g. Debian 8)
Debian 9.11

**Web server:** (e.g. Apache, Nginx,...)
Apache

**Database:** (e.g. MariaDB, SQLite or PostgreSQL)
MariaDB

**PHP version:** (e.g. 7.0.3)
PHP 7.2.27-1+0~20200123.34+debian9~1.gbp63c0bc

**Nextcloud Version:** (see admin page, e.g. 17.0.2)
17.0.2

**Updated from an older installed version or fresh install:**
Updated from an older installed version

**List of activated apps:**
Calendar and Contacts

```
Enabled:
  - accessibility: 1.3.0
  - activity: 2.10.1
  - admin_audit: 1.7.0
  - calendar: 2.0.1
  - cloud_federation_api: 1.0.0
  - comments: 1.7.0
  - contacts: 3.1.8
  - dav: 1.13.0
  - federatedfilesharing: 1.7.0
  - federation: 1.7.0
  - files: 1.12.0
  - files_pdfviewer: 1.6.0
  - files_rightclick: 0.15.1
  - files_sharing: 1.9.0
  - files_trashbin: 1.7.0
  - files_versions: 1.10.0
  - files_videoplayer: 1.6.0
  - firstrunwizard: 2.6.0
  - gallery: 18.4.0
  - logreader: 2.2.0
  - lookup_server_connector: 1.5.0
  - nextcloud_announcements: 1.6.0
  - notifications: 2.5.0
  - oauth2: 1.5.0
  - password_policy: 1.7.0
  - privacy: 1.1.0
  - provisioning_api: 1.7.0
  - recommendations: 0.5.0
  - serverinfo: 1.7.0
  - sharebymail: 1.7.0
  - support: 1.0.1
  - survey_client: 1.5.0
  - systemtags: 1.7.0
  - text: 1.1.1
  - theming: 1.8.0
  - twofactor_backupcodes: 1.6.0
  - updatenotification: 1.7.0
  - viewer: 1.2.0
  - workflowengine: 1.7.0
Disabled:
  - encryption
  - files_external
  - user_ldap
```

**Nextcloud configuration:**

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "example.com"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/example.com\/oc",
        "dbtype": "mysql",
        "version": "17.0.2.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "forcessl": true,
        "forceSSLforSubdomains": true,
        "defaultapp": "calendar",
        "theme": "",
        "maintenance": false,
        "loglevel": 1,
        "mail_smtpmode": "smtp",
        "trashbin_retention_obligation": "auto",
        "htaccess.RewriteBase": "\/oc",
        "mail_smtpauthtype": "LOGIN"
    }
}
```

#### Web server error log (e.g. /var/log/apache)
```
Unavailable
```

#### Log file (data/nextcloud.log)
```
Unavailable
```

#### Browser log
```
Unavailable
```

**Comment by georgehrke** [comment]
> In the NextCloud calendar app settings, change the time zone to a negative offset timezone, like America/Chicago (UTC-6).

I assume that the device you are viewing the calendar on is also set to America/Chicago? Is that correct?

**Comment by michaelray** [comment]
This issue is only happening in the NextCloud calendar app web UI.

I also use DAVx5 (DavDroid) to sync my calendar events to my Android phone calendar app, but I don't see this issue in the Android calendar app UI.

**Comment by michaelray** [comment]
If you're referring to the Windows 10 OS that I'm using to run Firefox to load the NextCloud calendar app web UI, then yes, it is also set to use America/Chicago time zone.

**Comment by michaelray** [comment]
I just went and checked in Windows 10 settings, and it actually doesn't label the time zone as "America/Chicago".  It labels it as "(UTC-6:00) Central Time (US & Canada)".  Those are both referring to the same time zone, of course, though.

**Comment by georgehrke** [comment]
I haven't been able to reproduce this so far, postponing to 2.0.3

**Comment by charlescurley** [comment]
I am seeing similar. The non-showing events, in my case, are non-repeating. It is the first day in a month's display. For December, 2019, the first of December is a Wednesday, and that displays fine, but Sunday November 29 does not display any events at all.

Similarly for week displays, and one day displays nothing.

This is on Vivaldi (a Chromium derivative) and Firefox on Debian Linux 10.3 as updated. Calendar is version 2.0.1. Nextcloud is version 17.0.3.

**Comment by michaelray** [comment]
It seems that this doesn't occur for me when I create a new event in the NextCloud calendar web UI after I've updated to version 2.0.1.  This specifically seems to occur only for events that were already in my calendar prior to the upgrade (and they didn't have this issue displaying in NC calendar version 1.x).

The calendar events created in the newer version of NC calendar appear to have quite a bit more additional data related to time zones that the other events don't have.

Here is sample calendar event data for a repeating calendar event that **is not** appearing on March 1, 2020 (it shows a PRODID of DAVdroid, even though I created this event in NC calendar version 1.x.. I'm not sure if DAVdroid automatically made changes to the event after I had originally created it in NC calendar 1.x or what, but I definitely didn't manually make any changes to the event in DAVdroid / Android Calendar app):

`<cal:calendar-data>BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//bitfire web engineering//DAVdroid 0.7.7 (ical4j 1.0.x)//EN
BEGIN:VEVENT
DTSTAMP:20150526T020933Z
UID:1869cf400f
SUMMARY:BILL due
LOCATION:BILL location\, online
DESCRIPTION:BILL description
TRANSP:TRANSPARENT
LAST-MODIFIED:20180120T115902
CLASS:PUBLIC
STATUS:CONFIRMED
DTSTART;VALUE=DATE:20140701
DTEND;VALUE=DATE:20140702
RRULE:FREQ=MONTHLY
END:VEVENT
END:VCALENDAR</cal:calendar-data>`

Here is sample calendar event data for a repeating calendar event that **is** appearing on March 1, 2020 (I recently added this one using NextCloud calendar 2.0.1, shortly after I did the upgrade):

`<cal:calendar-data>BEGIN:VCALENDAR
CALSCALE:GREGORIAN
VERSION:2.0
PRODID:-//IDN nextcloud.com//Calendar app 2.0.1//EN
BEGIN:VEVENT
CREATED:20200202T000534Z
DTSTAMP:20200202T000644Z
LAST-MODIFIED:20200202T000644Z
SEQUENCE:3
UID:f711bfde-2edb-4b19-9673-9f118b90f004
DTSTART;TZID=America/Chicago:20200201T000000
DTEND;TZID=America/Chicago:20200201T000000
SUMMARY:BILL due
LOCATION:BILL location\, online
DESCRIPTION:BILL description
TRANSP:TRANSPARENT
RRULE:FREQ=MONTHLY;BYMONTHDAY=1
END:VEVENT
BEGIN:VTIMEZONE
TZID:America/Chicago
BEGIN:DAYLIGHT
TZOFFSETFROM:-0600
TZOFFSETTO:-0500
TZNAME:CDT
DTSTART:19700308T020000
RRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=2SU
END:DAYLIGHT
BEGIN:STANDARD
TZOFFSETFROM:-0500
TZOFFSETTO:-0600
TZNAME:CST
DTSTART:19701101T020000
RRULE:FREQ=YEARLY;BYMONTH=11;BYDAY=1SU
END:STANDARD
END:VTIMEZONE
END:VCALENDAR</cal:calendar-data>`

**Comment by georgehrke** [comment]
@michaelray Thanks for the ics files, I have been able to reproduce it with those. This has already been fixed with https://github.com/georgehrke/calendar-js/pull/99 and will be fixed with Calendar 2.0.3.

> The calendar events created in the newer version of NC calendar appear to have quite a bit more additional data related to time zones that the other events don't have.

The first ics file doesn't contain any `VTIMEZONE` blocks, because the event is all-day and hence not associated with any timezone.
